src/Components/Server/src/Builder/RazorComponentsEndpointRouteBuilderExtensions.cs

```diff
@@ -0,0 +1,193 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Components;
+using Microsoft.AspNetCore.Components.Server;
+using Microsoft.AspNetCore.Routing;
+
+namespace Microsoft.AspNetCore.Builder
+{
+    /// <summary>
+    /// Extensions for <see cref="RazorComponentsEndpointBuilder"/>
+    /// </summary>
+    public static class RazorComponentsEndpointRouteBuilderExtensions
+    {
+        /// <summary>
+        /// Maps the SignalR Components Hub to the path <see cref="ComponentsHub.DefaultPath"/> and associates
```

Comment:
Could we singular-case all of these names? There's a mix of single and plural stuff and it's kinda annoying. My suggestion would be to prefer things like `ComponentHub` over `ComponentsHub`.

Reply:
I singular cased everything